statistics/handle/statscache.go

```diff
@@ -34,20 +34,13 @@ type statsCacheInner interface {
 	FreshMemUsage()
 	FreshTableCost(int64)
 	Copy() statsCacheInner
-}
-
-type cacheItem struct {
-	key   int64
-	value *statistics.Table
-	cost  int64
+	SetCapacity(int64)
 }
 
 func newStatsCache() statsCache {
+	capacity := variable.StatsCacheMemQuota.Load()
 	return statsCache{
-		statsCacheInner: &mapCache{
-			tables:   make(map[int64]cacheItem),
-			memUsage: 0,
-		},
+		statsCacheInner: newStatsLruCache(capacity),
```

Comment:
What about keeping mapCache so that we can fallback to mapCache if needed?

Reply:
updated.